src/server/pachyderm_test.go

```diff
@@ -7745,9 +7752,39 @@ func TestListDatumDuringJob(t *testing.T) {
 		}, backoff.NewTestingBackOff())
 	})
 
+	// initially since no datum chunks have been processed, we receive 0 datums
 	dis, err := c.ListDatumAll(pipelineJobInfo.PipelineJob.ID)
 	require.NoError(t, err)
-	require.Equal(t, 1, len(dis))
+	require.Equal(t, len(dis), 0)
+
+	// test job progress by waiting until some datums are returned, and verify that it's not all of them
+	require.NoErrorWithinT(t, 30*time.Second, func() error {
+		return backoff.Retry(func() error {
+			dis, err = c.ListDatumAll(pipelineJobInfo.PipelineJob.ID)
+			if err != nil {
+				return err
+			}
+			if len(dis) == 0 {
+				return errors.Errorf("expected some datums to be listed")
+			}
+			return nil
+		}, backoff.NewTestingBackOff())
+	})
+
+	// see that some datums have been processed
```

Comment:
I think the number of datums is necessarily >= 1 if we reach here since the retry errors until that is the case. It seems like we just need a check here to ensure that the number of datums recorded by the retry above are less than the total expected, and that the number of datums after the flush are equal to the total expected.